### PR TITLE
JSON CAPI - add strict option

### PR DESCRIPTION
**Pull request by oshadmi**
Changes corresponding to PR [#361](https://github.com/RedisJSON/RedisJSON/pull/361) in RedisJSON

### CODE REVIEW

**Comment by oshadmi** [comment]
@rafie Currently did not yet enable the `isStrict` option, which fails on `Array` and `Object`.

**Comment by oshadmi** [comment]
@rafie This change is too drastic, e.g.,  `test_json:testSearchUpdatedContent` in RedisJSON fails:
`[1L, ['labelT', 'rex']] == [1L, ['labelT', '"rex"']]`

Need to unquote strings only in specific cases - not clear which.

**Comment by ashtul** [comment]
Why do we need this?

File: src/rejson_api.h [hunk 1]
```diff
@@ -55,14 +55,18 @@ typedef struct RedisJSONAPI_V1 {
...
-    // FIXME: Handle null field
-    int (*getString)(RedisJSON json, const char **str, size_t *len);
-    int (*getStringFromKey)(RedisJSONKey key, const char *path, const char **str, size_t *len);
+    // Return the String for String
+    // Return the String representation for Numeric
+    // Return true or false for Bool
+    // Return the String representation for Array and Object if strict=0
+    // Return REDISMODULE_ERR for Array and Object if strict=1
+    // Return empty string for null
+    int (*getString)(RedisJSON json, const char **str, size_t *len, int isStrict);
```
Comment (MeirShpilraien) [inline src/rejson_api.h:64]: Why not return a string representation always? The user can check the type if he do not want to do it on array or object no?

**Review by MeirShpilraien — COMMENTED** [review]
Suggestion...

File: src/rejson_api.h (hunk 1 above)
Comment (oshadmi) [reply src/rejson_api.h:64]: When `isStrict=0` we always return a string representation.
The user can check the type or can also use `isStrict=1`

**Comment by oshadmi** [comment]
Replaced by other PR (related to RedisJSON PR #372)

**Pull request closed without merging**